ext/intl/idn/idn.c

```diff
@@ -128,18 +123,17 @@ static void php_intl_idn_handoff(INTERNAL_FUNCTION_PARAMETERS, int mode)
 		RETURN_THROWS();
 	}
 
-	if (variant != INTL_IDN_VARIANT_UTS46) {
-		php_intl_bad_args("invalid variant, must be INTL_IDNA_VARIANT_UTS46");
-		RETURN_FALSE;
-	}
-
-	if (ZSTR_LEN(domain) < 1) {
-		php_intl_bad_args("empty domain name");
-		RETURN_FALSE;
+	if (ZSTR_LEN(domain) == 0) {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
 	}
 	if (ZSTR_LEN(domain) > INT32_MAX - 1) {
-		php_intl_bad_args("domain name too large");
-		RETURN_FALSE;
+		zend_argument_value_error(1, "must be less than or equal to " PRId32 " bytes", INT32_MAX - 1);
```

Comment:
might be simpler to say lesser than INT32_MAX wdyt ?

Reply:
That actually is better :)

Reply:
Fixed in the same commit with a force push